File: Python/refer-offer/bu_ke_pai_zhong_de_shun_zi_lcof.py
from typing import *

def isStraight(nums: List[int]) -> bool:
    nums.sort()
    wang = 0
    for i in range(len(nums) - 1):
        if nums[i] == 0:
            wang += 1
        elif nums[i + 1] - 1 - nums[i] < 0:
            return False
        elif nums[i] != nums[i + 1] - 1:
            wang -= nums[i + 1] - 1 - nums[i]
    
    return True if wang >= 0 else False

if __name__ == "__main__":
    print(isStraight([0,12,11,11,0]))

File: Java/src/com/dbc/algorithm/ReferOffer/BuKePaiZhongDeShunZiLcof.java
package com.dbc.algorithm.ReferOffer;

import java.util.Arrays;

public class BuKePaiZhongDeShunZiLcof {
    public boolean isStraight(int[] nums) {
        Arrays.sort(nums);
        int count = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] == 0) count++;
            else if (nums[i + 1] - 1 - nums[i] < 0) {
                return false;
            } else {
                count -= nums[i + 1] - 1 - nums[i];
            }
        }
        return count >= 0;
    }
}
